packages/salesforcedx-vscode-core/src/commands/forceSourceDeploy.ts

```diff
@@ -5,37 +5,129 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import {
-  Command,
-  SfdxCommandBuilder
-} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { CliCommandExecutor, Command, SfdxCommandBuilder } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
 import { nls } from '../messages';
-import {
-  SfdxCommandlet,
-  SfdxCommandletExecutor,
-  SfdxWorkspaceChecker
-} from './commands';
-import {
-  FileType,
-  ManifestOrSourcePathGatherer,
-  SelectedPath
-} from './forceSourceRetrieve';
-
-export class ForceSourceDeployExecutor extends SfdxCommandletExecutor<
-  SelectedPath
-> {
+import { notificationService, ProgressNotification } from '../notifications';
+import { taskViewService } from '../statuses';
+import { SfdxCommandlet, SfdxCommandletExecutor, SfdxWorkspaceChecker } from './commands';
+import { FileType, ManifestOrSourcePathGatherer, SelectedPath } from './forceSourceRetrieve';
+
+interface Result {
+  columnNumber: string;
+  error: string;
+  filePath: string;
+  fullName: string;
+  lineNumber: string;
+  type: string;
+}
+
+interface ForceSourceDeployErrorResult {
+  message: string;
+  name: string;
+  result: Result[];
+  stack: string;
+  status: number;
+  warnings: any[];
+}
+
+// move this elsewhere?
+vscode.workspace.onDidChangeTextDocument((e => {
+  if (ForceSourceDeployExecutor._errorCollection.has(e.document.uri)) {
+    ForceSourceDeployExecutor._errorCollection.delete(e.document.uri);
+  }
+}));
+
+export class ForceSourceDeployExecutor extends SfdxCommandletExecutor<SelectedPath> {
+
+  public static _errorCollection = vscode.languages.createDiagnosticCollection('deploy-errors');
```

Comment:
Let's not use the `_` in front of variables names.

Reply:
whats your preferred naming convention for statics?